File: Cores/ExtremumScalaCore/src/main/scala/OSOL/Extremum/Core/Scala/Optimization/Testing/IntervalTester.scala
package OSOL.Extremum.Core.Scala.Optimization.Testing

import OSOL.Extremum.Core.Scala.Arithmetics.Interval
import OSOL.Extremum.Core.Scala.Arithmetics.Interval.Converters._
import OSOL.Extremum.Core.Scala.Optimization.Area
import OSOL.Extremum.Core.Scala.Vectors.IntervalVector

class IntervalTester extends Tester[IntervalVector, Interval, IntervalVector](
  testFunctions = IntervalTester.f,
  areas = IntervalTester.a,
  solutions = IntervalTester.s,
  tolerance = IntervalTester.tolerance,
  attempts = IntervalTester.attempts) {

}

object IntervalTester {

  val tolerance = 1e-3
  val attempts = 5

  val vars_1 = Seq("x")
  val vars_2 = Seq("x", "y")
  val vars_3 = Seq("x", "y", "z")


  val f1: Map[String, Interval] => Interval = (v: Map[String, Interval]) => v("x").pow(2.0)
  val a1: Area = vars_1.map(k => (k, (-10.0, 10.0))).toMap
  val s1: Map[String, Double] = vars_1.map(k => (k, 0.0)).toMap

  val f2: Map[String, Interval] => Interval = (v: Map[String, Interval]) => v("x").pow(2.0) + v("y").pow(2.0)
  val a2: Area = vars_2.map(k => (k, (-10.0, 10.0))).toMap
  val s2: Map[String, Double] = vars_2.map(k => (k, 0.0)).toMap

  val f3: Map[String, Interval] => Interval = (v: Map[String, Interval]) => v("x").pow(2.0) + v("y").pow(2.0) + v("z").pow(2.0)
  val a3: Area = vars_3.map(k => (k, (-10.0, 10.0))).toMap
  val s3: Map[String, Double] = vars_3.map(k => (k, 0.0)).toMap


  val f = Seq(f1, f2, f3)
  val a = Seq(a1, a2, a3)
  val s = Seq(s1, s2, s3)

}

File: Cores/OSOL.Extremum.Core.DotNet/Optimization/Testing/IntervalTester.cs
using System;
using System.Collections.Generic;
using System.Linq;
using OSOL.Extremum.Core.DotNet.Arithmetics;
using OSOL.Extremum.Core.DotNet.Vectors;

namespace OSOL.Extremum.Core.DotNet.Optimization.Testing
{
    public class IntervalTester: Tester<IntervalVector, Interval, IntervalVector>
    {

        public IntervalTester()
        {
            this.Tolerance = 1e-3;
            this.Attempts = 5;

            var vars_1 = new[] {"x"};
            var vars_2 = new[] {"x", "y"};
            var vars_3 = new[] {"x", "y", "z"};
            
            Func<Dictionary<string, Interval >, Interval> f1 = v => v["x"].Power(2.0);
            Dictionary<string, Tuple<double, double>> a1 = vars_1.ToDictionary(k => k, k => Tuple.Create(-10.0, 10.0));
            Dictionary<string, double> s1 = vars_1.ToDictionary(k => k, k => 0.0);
            
            Func<Dictionary<string, Interval >, Interval> f2 = v => v["x"].Power(2.0) + v["y"].Power(2.0);
            Dictionary<string, Tuple<double, double>> a2 = vars_1.ToDictionary(k => k, k => Tuple.Create(-10.0, 10.0));
            Dictionary<string, double> s2 = vars_2.ToDictionary(k => k, k => 0.0);
            
            Func<Dictionary<string, Interval >, Interval> f3 = v => v["x"].Power(2.0) + v["y"].Power(2.0) + v["z"].Power(2.0);
            Dictionary<string, Tuple<double, double>> a3 = vars_1.ToDictionary(k => k, k => Tuple.Create(-10.0, 10.0));
            Dictionary<string, double> s3 = vars_3.ToDictionary(k => k, k => 0.0);

            this.TestFunctions = new[] {f1, f2, f3};
            this.Areas = new[] {a1, a2, a3};
            this.Solutions = new[] {s1, s2, s3};

        }
        
    }
}